lib/scanner/index.js

```diff
@@ -43,9 +57,44 @@ class Scanner extends Emitter {
   stop() {
     debug('stop');
     MagnetScanner.stop();
+    this.stopExpireCheck();
     this.listener.remove();
   }
 
+  startExpireCheck() {
+    this.nextExpireCheck = setTimeout(() => {
```

Comment:
Yeah, we could do the check earlier, but we'd need a hook to kick the scanner to check for expiries again when the user changes that option. That's all possible though :)